contrib/gocql/gocql/gocql.go

```diff
@@ -111,17 +111,16 @@ func (tq *Query) newChildSpan(ctx context.Context) ddtrace.Span {
 }
 
 func (tq *Query) finishSpan(span ddtrace.Span, err error) {
-	if err == gocql.ErrNotFound {
-		// ErrNotFound is technically an error from gocql point of view.
+	if err != nil && tq.params.config.errCheck != nil && tq.params.config.errCheck(err) == false {
+		// If the error is explicitely marked as not-an-error, then skip and ignore.
+		// A typical use-case is gocql.ErrNotFound which is returned when scanning data,
+		// but no data is available so zero rows are returned.
 		//
-		// However, it is returned when "no data is available", even if
-		// the request was perfectly correct and executed well. It is
-		// only a marker of "zero rows returned".
-		//
-		// So APM graphs/data can expose a very high error rate if queries
-		// happen to try and select data which is just not there.
+		// This only affects whether the span/trace is marked as success/error,
+		// the calls to the gocql API still return the upstream error code.
```

Comment:
I think that the rest of this text maybe belongs in the documentation of `WithErrorCheck`